File: python/anagram.py
from sys import argv

from mylib import WORDS_DIR, jsonData, getWords, getAnagram

def anagram(str):
    """
    The main method to get the anagram
    """
    l = len(str)
    max = jsonData['MAX_CHAR']
    min = jsonData['MIN_CHAR']
    if l > max or l < min:
        print(f"Error: please provide a word between {min} and {max}")
    else:
        words = getWords(WORDS_DIR, l)
        return getAnagram(str, words)

def main(word):
    """
    wrapper method to hold everything together
    """
    result = anagram(word)
    # strange error: TypeError: 'bool' object is not subscriptable
    # if we do this: result[0] != False
    foundWord = result[0]
    if foundWord:
        print(f"We found an angram for {word} > {foundWord} after {result[1]}({result[2]}) try")
    else:
        print(f"Sorry could not find anything, after {result[1]}({result[2]}) try")


if __name__ == '__main__':
    script, word = argv
    main(word)

File: php/anagram.php
<?php
chdir( dirname ( __FILE__ ) );
require "./mylib.php";

/**
 * The main method to get the anagram
 *
 */
function anagram($str) {
  $l = strlen($str);
  $max = $jsonData["MAX_CHAR"];
  $min = $jsonData["MIN_CHAR"];

  if ($l > $max || $l < $min) {
    echo "Error: please proide a word between $min and $max";
  } else {
    $words = getWords();

    return getAnagram($str, $words);
  }
}


/**
 * The main method for the command line interface
 *
 */
function main($word) {
  $result = anagram($word);
  if ($result) {
    echo "We found an anagram for $word > $result";
  } else {
    echo "Sorry could not find anything";
  }
}

// The $argv will present even the command line include this script
// therefore we need to add a variable from the callee to identify it
if (!defined('IN_SCRIPT') && $argv) {
  // should also check if there is a 1
  if (count($argv) > 1) {
    main($argv[1]);
  } else {
    echo "You need to provide a word";
  }
}
?>
